src/TypeDefinition.js

```diff
@@ -92,12 +93,11 @@ export type NavigationRouter = {
 };
 
 export type NavigationScreenOption<T> =
+  | T
   | (navigation: NavigationScreenProp<NavigationRoute, NavigationAction>,
-    config: NavigationScreenOptionConfig,
-    router?: NavigationRouter) => T
-  | T;
+    config: T) => T;
 
-export type Style = Object | number | false | void;
+export type Style = { [key: string]: any } | number | false | null | void | Array<Style>;
```

Comment:
Nice! StyleSheets are crazy but I think you may have nailed it. Maybe we could get RN core to export this type?